src/groupmod.c

```diff
@@ -250,7 +250,7 @@ static void grp_update (void)
 
 		if (!aflg) {
 			// requested to replace the existing groups
-			if (NULL != grp.gr_mem[0])
+			if ((NULL != grp.gr_mem[0]) && !nflg)
```

Comment:
This doesn't fix the following reproducer:

```sh
$ sudo cat /etc/group | grep g2
g2:x:1008:u1,u2,u3
$ sudo groupmod -U u1,u2 g2
Segmentation fault
```

Reply:
I have two ways to modify it:

One is to directly modify the dangerous assignment method of `grp=* ogrp`, but this may affect other existing replacement mechanisms (I have not delved into the specifics yet)

Another way is to directly delete the first free. That is, `gr_free_members (&grp)`. According to the previous tracking, we can clearly see that for the `-U` option, a free is always triggered within `gr_update`, while `-n & -U` will be triggered once in remove (and why `-n & -U` did not trigger free in `gr_update` is because it created a new entry rather than modifying the old one)

I will make the modifications later and use more test cases to verify these modifications